src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/McpServer/McpServer-CSharp/Tools/RandomNumberTools.cs

```diff
@@ -0,0 +1,49 @@
+using System.ComponentModel;
+using ModelContextProtocol.Server;
+
+namespace McpServer_CSharp.Tools;
+
+/// <summary>
+/// Sample MCP tools for demonstration purposes.
+/// These tools can be invoked by MCP clients to perform various operations.
+/// </summary>
+public class RandomNumberTools
+{
+    private readonly int _maxNumber;
+
+    public RandomNumberTools()
+    {
+        // Process configuration settings from the environment variables.
+        // These will be provided by the MCP client application, such as VS Code.
+        // Configuration settings could be provided via dependency injection and the IOptions pattern.
+        var maxNumberEnv = Environment.GetEnvironmentVariable("MAX_RANDOM_NUMBER");
+        if (!int.TryParse(maxNumberEnv, out var maxNumber) || maxNumber <= 0)
+        {
+            throw new InvalidOperationException("Error: you must set the MAX_RANDOM_NUMBER environment variable to a positive integer.");
+        }
+
+        _maxNumber = maxNumber;
+    }
+
+    /// <summary>
+    /// Returns the maximum random number that can be generated by the tool.
+    /// </summary>
+    /// <returns>The maximum random number.</returns>
+    [McpServerTool(Name = "get_max_random_number")]
```

Comment:
Once https://github.com/modelcontextprotocol/csharp-sdk/pull/568 is merged, this is the name that would be generated for this method, anyway. You can still include it, but it's a nop.

Reply:
Cool. I'll leave it for now, so that the README names/screenshots match up when we refactor later (new SDK version).